src/main/java/software/amazon/awssdk/crt/mqtt/MqttClientConnection.java

```diff
@@ -106,6 +117,159 @@ protected boolean canReleaseReferencesImmediately() {
         return false;
     }
 
+    /**
+     * Configures the connection-related callbacks to use on this connection
+     *
+     * @param callbacks connection event callbacks to use
+     */
+    public void setConnectionCallbacks(MqttClientConnectionEvents connectionCallbacks) {
+        this.userConnectionCallbacks = connectionCallbacks;
+    }
+
+    /**
+     * Queries the connection-related callbacks being used by this connection
+     *
+     * @return the connection event callbacks being used
+     */
+    public MqttClientConnectionEvents getConnectionCallbacks() {
+        return userConnectionCallbacks;
+    }
+
+    /**
+     * Configures the client_id to use with this connection
+     *
+     * @param clientId The client id for this connection. Needs to be unique across
+     *                  all devices/clients.
+     */
+    public void setClientId(String clientId) {
```

Comment:
The way we've done this in the other builders was to  have these params be part of the constructor, and have the builder invoke the constructor once it had all the params. By making this public, it makes the API to this class VERY confusing. I would suggest making this private or something if it must be done this way, but I'd rather use a constructor like we do elsewhere.

I'd much rather keep the config object and have the builder close it when it's done creating a connection

Reply:
Done